src/core/display.cpp

```diff
@@ -1252,3 +1255,195 @@ uint16_t getColorVariation(uint16_t color, int delta, int direction) {
+    bmpFS.close();
+    return false;
+  }
+  bmpFS.close();
+  return true;
+}
+
+bool IRAM_ATTR drawImg(FS fs, String filename, int x, int y, bool center, int playDurationMs) {
+  String ext = filename.substring(filename.lastIndexOf('.'));
+  ext.toLowerCase();
+  if(ext.endsWith("jpg")) return showJpeg(fs,filename,x,y,center);
+  else if(ext.endsWith("bmp")) return drawBmp(fs,filename,x,y,center);
+  else if(ext.endsWith("png")) return drawPNG(fs,filename,x,y,center);
+  else if(ext.endsWith("gif")) return showGif(&fs, filename.c_str(),x,y,center,playDurationMs);
+  else log_e("Image not supported");
+  
+  return false;
+}
+
+#if !defined(LITE_MODE)
+/// Draw PNG files
+
+#include <PNGdec.h>
+#define MAX_IMAGE_WIDTH 320
+PNG* png;
+// Functions to access a file on the SD card
+File myfile;
+FS* _fs;
+
+void * IRAM_ATTR myOpen(const char *filename, int32_t *size) {
+  //Serial.printf("Attempting to open %s\n", filename);
+  myfile = _fs->open(filename);
+  *size = myfile.size();
+  return &myfile;
+}
+void IRAM_ATTR myClose(void *handle) {
+  if (myfile) myfile.close();
+}
+int32_t IRAM_ATTR myRead(PNGFILE *handle, uint8_t *buffer, int32_t length) {
+  if (!myfile) return 0;
+  return myfile.read(buffer, length);
+}
+int32_t IRAM_ATTR mySeek(PNGFILE *handle, int32_t position) {
+  if (!myfile) return 0;
+  return myfile.seek(position);
+}
+// Function to draw pixels to the display
+int16_t xpos = 0;
+int16_t ypos = 0;
+void IRAM_ATTR PNGDraw(PNGDRAW *pDraw) {
+  uint16_t usPixels[320];
+  //static uint16_t dmaBuffer[MAX_IMAGE_WIDTH]; // static so buffer persists after fn exit
+  uint8_t r = ((uint16_t)bruceConfig.bgColor & 0xF800) >> 8;
+  uint8_t g = ((uint16_t)bruceConfig.bgColor & 0x07E0) >> 3;
+  uint8_t b = ((uint16_t)bruceConfig.bgColor & 0x001F) << 3;
+  png->getLineAsRGB565(pDraw, usPixels, PNG_RGB565_BIG_ENDIAN, b << 16 | g << 8 | r);
+  tft.drawPixel(0,0,0);
+  tft.drawPixel(0,0,0);
+  tft.pushImage(xpos, ypos + pDraw->y, pDraw->iWidth, 1, usPixels);
+}
+
+bool IRAM_ATTR drawPNG(FS fs, String filename, int x, int y, bool center) {
+  if ((x >= tft.width()) || (y >= tft.height())) return false;
+  _fs = &fs;
+  uint32_t dt = millis();
+  png = new PNG();
+  int16_t rc = png->open(filename.c_str(), myOpen, myClose, myRead, mySeek, PNGDraw);
+  if (rc == PNG_SUCCESS) {
+    //Serial.printf("image specs: (%d x %d), %d bpp, pixel type: %d\n", png->getWidth(), png->getHeight(), png->getBpp(), png->getPixelType());
+    
+    if(center) {
+      xpos=x+(tftWidth-png->getWidth())/2;
+      ypos=y+(tftHeight-png->getHeight())/2;
+    }
+    
+    if (png->getWidth() > MAX_IMAGE_WIDTH) {
+      Serial.println("Image too wide for allocated line buffer size!");
+    }
+    else {
+      rc = png->decode(NULL, 0);
+      png->close();
+    }
+    // How long did rendering take...
+    Serial.print("PNG Loaded in "); Serial.print(millis()-dt); Serial.println("ms");
+  }
+  else {
+    ERROR:
+    return false;
+  }
+  return true;
+
+}
```

Comment:
there is memory leak here, no `delete png;`

```suggestion
    if (png->getWidth() > MAX_IMAGE_WIDTH) {
      log_e("Image too wide for allocated line buffer size!");
    } else {
      png->decode(NULL, 0);
    }
    png->close();
    delete png;
    // How long did rendering take...
    log_d("PNG Loaded in %d ms", Serial.print(millis() - dt));
  } else {
    delete png;
    return false;
  }
  return true;

}
```